src/Database/PostgreSQL/Simple/Range.hs

```diff
@@ -49,6 +49,7 @@ import           Data.Word                            (Word, Word16, Word32,
 
 import           Database.PostgreSQL.Simple.Compat    (scientificBuilder, (<>), toByteString)
 import           Database.PostgreSQL.Simple.FromField
+import           Database.PostgreSQL.Simple.Internal (setTypeOid)
 import           Database.PostgreSQL.Simple.Time
```

Comment:
Honestly,  I think `setTypeOid`  (and maybe some other things to manipulate the "unpacked" stuff?) should be made part of the public interface.